Assets/Scripts/CombatZone.cs

```diff
@@ -33,50 +38,97 @@ public void Start()
 
     }
 
+    public void Update()
+    {
+        if (!Timer.IsInProgress())
+            return;
+
+        if (CombatStarted)
+            DisablePlayerMovement();
```

Comment:
Why are you doing this in the update?

Reply:
Fixed. Moved it to the OnTriggerEnter function.